src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -150,9 +161,45 @@ def find_scm(pom: Element, tags: list[str]) -> tuple[Iterator[str], int]:
                 # Add the contents of the final tag
                 results.append(element.text.strip())
 
+    # Resolve any Maven properties within the results
+    if resolve_properties:
+        results = _resolve_properties(pom, results)
+
     return iter(results), len(results)
 
 
+def _resolve_properties(pom: Element, values: list[str]) -> list[str]:
+    """Resolve any Maven properties found within the passed list of values.
+
+    Maven POM files have five different use cases for properties (see https://maven.apache.org/pom.html).
+    Only the two that relate to contents found elsewhere within the same POM file are considered here.
+    That is: ${project.x} where x can be a child tag at any depth, or ${x} where x is found at project.properties.x
+    Entries with rejected properties are removed from the value list.
+    """
+    resolved_values = []
+    for value in values:
+        replacements: list = []
+        # Calculate replacements
+        for match in re.finditer("\\$\\{[^}]+}", value):
+            text = match.group().replace("$", "").replace("{", "").replace("}", "")
+            if text.startswith("project."):
+                text = text.replace("project.", "")
+            else:
+                text = f"properties.{text}"
+            value_iterator, count = find_scm(pom, [text], False)
```

Comment:
Looks like @tromai has raised the same [issue](https://github.com/oracle/macaron/pull/271#discussion_r1222351978) :smile:

Reply:
Since the previous comment I added a boolean flag that disables property resolution after the first attempt. It is also included in the tests. If preferable, the boolean flag could changed to a simple counter representing depth of recursion.

Reply:
> Since the previous comment I added a boolean flag that disables property resolution after the first attempt. It is also included in the tests. If preferable, the boolean flag could changed to a simple counter representing depth of recursion.

Right, I can understand it now. A comment to explicitly show the flag is turned off to avoid recursion would be good enough.